### PR TITLE
curvefs tools space-status

### PR DESCRIPTION
**Pull request by Cyber-SiKu**
### What problem does this PR solve?

Issue Number: close #xxx 

Problem Summary:

### What is changed and how it works?

What's Changed:

How it Works:

Side effects(Breaking backward compatibility? Performance regression?): 

### Check List

- [ ] Relevant documentation/comments is changed or added
- [ ] I acknowledge that all my contributions will be made under the project's license

### CODE REVIEW

File: curvefs/src/tools/build/curvefs_build_topology_tool.cpp [hunk 1]
```diff
@@ -42,8 +42,8 @@ void UpdateFlagsFromConf(curve::common::Configuration* conf) {
         LOG(INFO) << "conf: " << FLAGS_mds_addr;
     }
 
-    if (GetCommandLineFlagInfo("rpcTimeOutMs", &info) && info.is_default) {
-        conf->GetUInt32Value("rpcTimeoutMs", &FLAGS_rpcTimeOutMs);
+    if (GetCommandLineFlagInfo("rpcTimeOut", &info) && info.is_default) {
+        conf->GetUInt32Value("rpcTimeout", &FLAGS_rpcTimeOutMs);
```
Comment (SeanHai) [inline curvefs/src/tools/build/curvefs_build_topology_tool.cpp:46]: Why change to rpcTimeOut ? The conf file is rpcTimeoutMs

File: curvefs/src/tools/curvefs_tool.h [hunk 2]
```diff
@@ -25,43 +25,43 @@
...
 #include <string>
+#include <utility>
 #include <vector>
 
 #include "curvefs/src/tools/curvefs_tool_define.h"
+#include "curvefs/src/tools/curvefs_tool_metric.h"
 #include "src/common/configuration.h"
 
 DECLARE_string(confPath);
+DECLARE_uint64(rpcTimeout);
```
Comment (SeanHai) [inline curvefs/src/tools/curvefs_tool.h:46]: Add unit （rpcTimeoutMS or rpcTimeoutSec ...）

File: curvefs/src/tools/curvefs_tool_define.cpp [hunk 3]
```diff
@@ -21,12 +21,21 @@
...
+              "metaserver addr");
+DEFINE_string(etcdAddr, "127.0.0.1:2379", "etcd addr");
 DEFINE_bool(example, false, "print the example of usage");
-DEFINE_string(confPath, "curvefs/conf/tools.conf", "config file path of tools");
+DEFINE_string(confPath, "/etc/curvefs/tools.conf", "config file path of tools");
 DEFINE_string(fsname, "curvefs", "fs name");
+DEFINE_string(fsId, "1,2,3", "fs id");
 DEFINE_string(mountpoint, "127.0.0.1:/mnt/curvefs-umount-test",
               "curvefs mount in local path");
+DEFINE_uint64(rpcTimeout, 3000, "millisecond for rpc timeout");
```
Comment (SeanHai) [inline curvefs/src/tools/curvefs_tool_define.cpp:34]: Add unit to rpcTimeout

File: curvefs/src/tools/curvefs_tool_define.h [hunk 4]
```diff
@@ -26,48 +26,95 @@
...
-    "Usage: curve_ops_tool [Command] [OPTIONS...]\n"
+    "Usage: curvefs_tool [Command] [OPTIONS...]\n"
     "COMMANDS:\n"  // NOLINT
     "version: show the version of cluster\n"
     "build-topology: build cluster topology based on topo.json\n"
-    "umountfs: umount curvefs from local and cluster"
+    "umount-fs: umount curvefs from local and cluster\n"
+    "usage-metadata: show the metadata usage of cluster\n"
+    "status-mds: show the status of mds\n"
+    "status-metaserver-: show the status of metaserver\n"
```
Comment (SeanHai) [inline curvefs/src/tools/curvefs_tool_define.h:82]: status-metaserver- not status-metaserver ?

File: curvefs/proto/metaserver.proto [hunk 5]
```diff
@@ -294,6 +294,14 @@ message MetaServerMetadata {
     // required uint32 checksum = 4;
 };
 
+message GetAllCopysetsIdRequest {
+};
```
Comment (wu-hanqing) [inline curvefs/proto/metaserver.proto:298]: `;` is redundant

File: curvefs/proto/metaserver.proto [hunk 6]
```diff
@@ -294,6 +294,14 @@ message MetaServerMetadata {
...
+
+message GetAllCopysetsIdResponse {
+    required MetaStatusCode statusCode = 1;
+    repeated uint32 copysetsId = 2;
+};
```
Comment (wu-hanqing) [inline curvefs/proto/metaserver.proto:303]: ditto

File: curvefs/src/tools/curvefs_tool_define.cpp [hunk 7]
```diff
@@ -50,5 +59,123 @@ void SetFlagInfo(curve::common::Configuration* conf,
...
+std::function<void(curve::common::Configuration*, google::CommandLineFlagInfo*)>
+    SetMetaserverAddr = std::bind(&SetFlagInfo<fLS::clstring>,
+                                  std::placeholders::_1, std::placeholders::_2,
+                                  "metaserverAddr", &FLAGS_metaserverAddr);
+
+std::function<void(curve::common::Configuration*, google::CommandLineFlagInfo*)>
+    SetEtcdAddr = std::bind(&SetFlagInfo<fLS::clstring>, std::placeholders::_1,
+                            std::placeholders::_2, "etcdAddr", &FLAGS_etcdAddr);
+
+auto StrVec2Str(const std::vector<std::string>& strVec) -> std::string {
```
Comment (wu-hanqing) [inline curvefs/src/tools/curvefs_tool_define.cpp:82]:
```cpp
std::string StrVec2Str(....
```

and same as below functions

File: curvefs/src/tools/query/curvefs_copyset_query.cpp [hunk 8]
```diff
@@ -0,0 +1,179 @@
...
+        }
+    }
+    return ret;
+}
+
+std::map<uint32_t, std::vector<StatusType>> CopysetQueryTool::GetId2Status() {
+    return id2Status_;
+}
+
+bool CopysetQueryTool::SplitPeerId(const std::string& peerId,
```
Comment (wu-hanqing) [inline curvefs/src/tools/query/curvefs_copyset_query.cpp:166]: why copy this function? put previous function into common, and directly use it

File: curvefs/src/tools/space/curvefs_space_base_tool.cpp [hunk 9]
```diff
@@ -0,0 +1,56 @@
...
+
+#include "curvefs/src/tools/space/curvefs_space_base_tool.h"
+
+namespace curvefs {
+namespace tools {
+namespace space {
+
+const uint32_t TB = 1024 * 1024 * 1024;
+const uint32_t GB = 1024 * 1024;
+const uint32_t MB = 1024;
```
Comment (wu-hanqing) [inline curvefs/src/tools/space/curvefs_space_base_tool.cpp:31]: define as uint64_t is better

File: curvefs/test/tools/curvefs_space_base_tool_test.cpp [hunk 10]
```diff
@@ -0,0 +1,58 @@
...
+    ASSERT_EQ(ToReadableByte(2 * 1024 + 1), "2.00 MB");
+}
+
+TEST_F(BaseToolTest, GB_test) {
+    ASSERT_EQ(ToReadableByte(3 * 1024 * 1024 + 1023 * 1024), "4.00 GB");
+}
+
+TEST_F(BaseToolTest, TB_test) {
+    uint64_t byte =
+        (uint64_t)100 * (uint64_t)1024 * (uint64_t)1024 * (uint64_t)1024;
```
Comment (wu-hanqing) [inline curvefs/test/tools/curvefs_space_base_tool_test.cpp:52]: `100ull * 1024 * 1024 * 1024`

File: curvefs/src/tools/build/curvefs_build_topology_tool.cpp (hunk 1 above)
Comment (Cyber-SiKu) [reply curvefs/src/tools/build/curvefs_build_topology_tool.cpp]: > Why change to rpcTimeOut ? The conf file is rpcTimeoutMs

fix

File: curvefs/src/tools/curvefs_tool.h (hunk 2 above)
Comment (Cyber-SiKu) [reply curvefs/src/tools/curvefs_tool.h]: > Add unit （rpcTimeoutMS or rpcTimeoutSec ...）

fix

File: curvefs/src/tools/curvefs_tool_define.cpp (hunk 3 above)
Comment (Cyber-SiKu) [reply curvefs/src/tools/curvefs_tool_define.cpp]: > Add unit to rpcTimeout

fix

File: curvefs/src/tools/curvefs_tool_define.h (hunk 4 above)
Comment (Cyber-SiKu) [reply curvefs/src/tools/curvefs_tool_define.h:82]: > status-metaserver- not status-metaserver ?

fix

File: curvefs/proto/metaserver.proto (hunk 5 above)
Comment (Cyber-SiKu) [reply curvefs/proto/metaserver.proto:298]: > `;` is redundant

fix

File: curvefs/proto/metaserver.proto (hunk 6 above)
Comment (Cyber-SiKu) [reply curvefs/proto/metaserver.proto:303]: > ditto

fix

File: curvefs/src/tools/curvefs_tool_define.cpp (hunk 7 above)
Comment (Cyber-SiKu) [reply curvefs/src/tools/curvefs_tool_define.cpp:81]: fix

File: curvefs/src/tools/query/curvefs_copyset_query.cpp (hunk 8 above)
Comment (Cyber-SiKu) [reply curvefs/src/tools/query/curvefs_copyset_query.cpp:166]: > why copy this function? put previous function into common, and directly use it

fix

File: curvefs/src/tools/space/curvefs_space_base_tool.cpp (hunk 9 above)
Comment (Cyber-SiKu) [reply curvefs/src/tools/space/curvefs_space_base_tool.cpp:31]: > define as uint64_t is better

fix

File: curvefs/test/tools/curvefs_space_base_tool_test.cpp (hunk 10 above)
Comment (Cyber-SiKu) [reply curvefs/test/tools/curvefs_space_base_tool_test.cpp]: > `100ull * 1024 * 1024 * 1024`

fix

**Comment by Cyber-SiKu** [comment]
recheck

File: src/mds/topology/topology_item.h [hunk 11]
```diff
@@ -861,6 +861,8 @@ bool SplitPeerId(
     uint32_t *port,
     uint32_t *idx = nullptr);
 
+bool SplitPeerId(const std::string& peerId, std::string* addr);
```
Comment (SeanHai) [inline src/mds/topology/topology_item.h:864]: You can add this function in 'curvefs/src/mds/topology/deal_peerid.h' or use function 'SplitPeerId' in it directly.

Comment (Cyber-SiKu) [reply src/mds/topology/topology_item.h:864]: > You can add this function in 'curvefs/src/mds/topology/deal_peerid.h' or use function 'SplitPeerId' in it directly.

fix

**Review by wu-hanqing — APPROVED** [review]

**Review by ilixiaocui — APPROVED** [review]

**Pull request merged**